### ISSUE TITLE
The score of the best model is different between trial.json and model.evaluate()

### ISSUE THREAD

**Issue by vincentluo91** [keras-team/autokeras#933]
Hi, I found the accuracy(0.737) returned by model.evaluate(x_val,y_val)  is different with the score(0.764) stored in the trial.json file(the higest score trial file). Is there something wrong with my understanding ? Any help would be appreciated. 

Here is the code:
auto_model = ak.AutoModel(
    inputs=input_node, 
    outputs=output_node,
    max_trials=100,
    directory="autokeras_dense",
    objective='val_accuracy',
)
H=auto_model.fit(x_train, y_train,validation_data=(x_val,y_val),epochs=40,verbose=0)

print(auto_model.evaluate(x_val, y_val))

The version information and I use GPU:
python --version Python 3.6.9 .
tf.version '2.1.0'
tf.keras.version '2.2.4-tf'
autokeras-1.0.0.

**Comment by haifeng-jin** [comment]
@vincentluo91 I think this is normal behavior. The value recorded in the json file is the highest score through all the epochs instead of the final score.
